### DISCUSSION TITLE
Development thoughts for adding parameters to do vegeonly in batch mode

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#25] [General]
I started to test things in a branch to add a parameter that forces the script to do vegetation generation only in batch mode. The approach is as simple as possible by:

1. adding a parameter in the .ini-fil:
```
# vegetation lony in batch mode, normal processing with countours = 0, only do vegetation in batch mode = 1
vegeonly=1

```
2. in the perl script add a line that reads and assigns the value from the .ini-file. Around lin 360 I added:
`$vegeonly = 0;$Config->{_}->{vegeonly};
`

3. adding some if-statements in the perl script to exclude certain processes like creating contours and cliffs, which takes a lot of time. E.g. like this:
```
if ( $vegeonly == 1 ) {
## cliff maker
    print "\nCliff generation \n";

    if ( $pcliff3 > 0 && $pcliff2 > 0 ) {
        print "Old cliff3 parameter found, using cliff3 instead of cliff2...\n";
        $pcliff2 = $pcliff3;
    }
    system("rusty-pullauta $thread makecliffs");
## cliff generalizer
    #    print "\nCliff generation part 2\n";
    #    system("perl pullauta $thread cliffgeneralize $psteepness");
}
```

This works for most parts and I have managed to exclude the making of cliffs and the standard contours and dot knolls.

But when I try to also exclude the first section, which creates the 03-contours, I later get errors in the vegetation creation process. I assume there are files being created in the first contour creation process that later is used when doing vegetation.

Does anyone have know how this works and have an idea of how to solve this?

The problem happens when I exclude these lines around row 2060:
```
## 2m grid & controus 0.3
    print "\nKnoll detection part 1\n";
    system(
        "rusty-pullauta $thread xyz2contours ".(0.3*$scalefactor)." xyztemp.xyz xyz_03.xyz contours03.dxf ground"
    );
```

**Comment by rphlo** [comment]
I am a bit busy right now but I can take a look either this weekend or during next week.
Do you have a branch on github with your current progress, and if not could you upload one?

**Comment by stefankinell** [comment]
I have a branch at https://github.com/stefankinell/rusty-pullauta/ where I just uploaded the first changes I did.

**Reply by rphlo** [reply]
Will see what I can do

**Comment by rphlo** [comment]
Proposed a pull request to your master branch that avoid useless `contours03.dxf` computation with vegeonly mode.

**Reply by stefankinell** [reply]
I looked it. And you are correct, the 03-contours are not generated now. 

But the strange thing is that it actually took longer time (just a few seconds) to run the script with these changes than it took before. I still have formlines created and the script output indicates temp polylines are produced:

```
Running xyz2contours  0.3 xyztemp.xyz xyz_03.xyz null true
Generating temp polylines: 5%
Generating temp polylines: 11%
Generating temp polylines: 16%
Generating temp polylines: 22%
``` 

I'll look more into it and see if I can find out how to make it work. And to a test run with a larger batch to see if it makes a difference.

**Reply by rphlo** [reply]
did you run 'cargo buid --release'

**Reply by rphlo** [reply]
and copy binary to a $PATH directory

**Reply by stefankinell** [reply]
My bad. I did not realize you made changes to main.rs as well. I now ran build and it removed polyline creation, and shaved of a few seconds more. 

I'll do a test on a full set tonight to verify, and then ask for a pull request to your main code.

**Comment by stefankinell** [comment]
I tried it some more and from what I have seen it works OK. So I tried doing my first pull request towards your main branch.

**Reply by rphlo** [reply]
I merged the changes but not from your pull request as it contained changes in README and pullauta.ini that I didnt want included

**Reply by stefankinell** [reply]
Good, I realized I made a mistake after having sent in the pull request. Thanks for noticing it.
